### PR TITLE
Migration: Android 12 Unsafe Intent launches 

### LINKED ISSUE #16093: Android 12: Security and performance - Unsafe intent launches
Parent #16062

[Unsafe intent launches](https://developer.android.com/about/versions/12/behavior-changes-12#unsafe-intent-launches)

The goal of this task is to detect and fix unsafe launches from the app.

Found atleast two unsafe launches from the app: 

**From Login Lib** 
```
D/StrictMode: StrictMode policy violation: android.os.strictmode.UnsafeIntentLaunchViolation: Launch of unsafe intent: Intent { (has extras) }
        at android.os.StrictMode.onUnsafeIntentLaunch(StrictMode.java:2330)
        at android.content.Intent.prepareToLeaveProcess(Intent.java:11450)
        at android.content.Intent.prepareToLeaveProcess(Intent.java:11368)
        at android.app.Activity.startIntentSenderForResultInner(Activity.java:5654)
        at android.app.Activity.autofillClientAuthenticate(Activity.java:8546)
        at android.view.autofill.AutofillManager.authenticate(AutofillManager.java:2224)
        at android.view.autofill.AutofillManager.access$2500(AutofillManager.java:176)
        at android.view.autofill.AutofillManager$AutofillManagerClient.lambda$authenticate$3(AutofillManager.java:3506)
        at android.view.autofill.AutofillManager$AutofillManagerClient$$ExternalSyntheticLambda1.run(Unknown Source:12)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**EditPostActivity**

```
D/StrictMode: StrictMode policy violation: android.os.strictmode.UnsafeIntentLaunchViolation: Launch of unsafe intent: Intent { cmp=org.wordpress.android.prealpha/org.wordpress.android.ui.posts.EditPostActivity (has extras) }
        at android.os.StrictMode.onUnsafeIntentLaunch(StrictMode.java:2330)
        at android.content.Intent.prepareToLeaveProcess(Intent.java:11450)
        at android.content.Intent.prepareToLeaveProcess(Intent.java:11368)
        at android.app.Activity.finish(Activity.java:6463)
        at android.app.Activity.finish(Activity.java:6499)
        at org.wordpress.android.ui.posts.EditPostActivity.lambda$startObserving$8$EditPostActivity(EditPostActivity.java:898)
        at org.wordpress.android.ui.posts.-$$Lambda$EditPostActivity$0mYQlvvtdaW0SjEuv-bQkUzqaXY.invoke(Unknown Source:4)
        at org.wordpress.android.viewmodel.Event.applyIfNotHandled(Event.kt:26)
        at org.wordpress.android.ui.posts.EditPostActivity.lambda$startObserving$9$EditPostActivity(EditPostActivity.java:884)
        at org.wordpress.android.ui.posts.-$$Lambda$EditPostActivity$SR4BjYkthqVIkINAm1tyLY480sk.onChanged(Unknown Source:4)
        at androidx.lifecycle.LiveData.considerNotify(LiveData.java:133)
        at androidx.lifecycle.LiveData.dispatchingValue(LiveData.java:151)
        at androidx.lifecycle.LiveData.setValue(LiveData.java:309)
        at androidx.lifecycle.MutableLiveData.setValue(MutableLiveData.java:50)
        at androidx.lifecycle.LiveData$1.run(LiveData.java:93)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7839)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

### PR DESCRIPTION
**Pull request by AjeshRPai**
Part of #16093 

This PR makes the necessary changes for handling unsafe intent launch `EditPostActivity.java`.

Note: In order to find out if there are any other activities which are launched unsafely, we can run test 2. 

#### To test:

Pre-requisites 
To test this change, add the following code to the app start up and run the app on an Android 12 device. 
```
StrictMode.setVmPolicy(
                VmPolicy.Builder()
                // Other StrictMode checks that you've previously added.
                // ...
                .detectUnsafeIntentLaunch()
                .penaltyLog()
                .penaltyDeath()
                // Consider also adding penaltyDeath()
                .build())
```

Alternatively revert this commit 6036084955400ae983bec0c6390fb0fb28a2b57b

### Test 1
- Launch app 
- Navigate to my site -> posts  
- Edit post from all the tabs draft/scheduled/published
- Verify that the post is updated and everything is working as expected

### Test 2 
A smoke testing of the entire app to launch all the activities might be required to discover the unsafe launches. Also, deep-link navigation launches should be tested for sanity.  

## Regression Notes
1. Potential unintended areas of impact
Edit Posts is not working as expected 

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual tests

3. What automated tests I added (or what prevented me from doing so)
N/A

PR submission checklist:
- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java [hunk 1]
```diff
@@ -2006,7 +2006,7 @@ private boolean isNewPost() {
     }
 
     private void saveResult(boolean saved, boolean uploadNotStarted) {
-        Intent i = getIntent();
+        Intent i = new Intent();
```
Comment (ashiagr) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java:2009]: By replacing `getIntent()` to `new Intent()` we'll be losing already existing intent extra data (e.g. `SITE`, `postModelLocalId`):

intent extra

which might be [needed by the calling activity's `onActivityResult()` ](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java#L1141-L1154).

Some of the [suggestions](https://developer.android.com/guide/components/intents-filters#UseIntentsMoreResponsibly) we can try:

- Copy only the essential extras within intents, and perform any necessary sanitation and validation.
   To the new `Intent`, you can copy intent extra data that might be needed by the calling activities. 

- Don't export your app's components unnecessarily. 
   Test that `EditPostActivity` is not invoked from any external process and mark `exported=false` in `AndroidManifest`.

**Review by ashiagr — CHANGES_REQUESTED** [review]
👋 @AjeshRPai, thank you for giving it a try but looks like we need to be a bit careful here. I left a note with a potential issue in the solution. 

Hope it makes sense. We can discuss it if you need any help.

**Comment by RenanLukas** [comment]
Hey @AjeshRPai and @ashiagr  👋 
I've submitted a PR with a possible fix for this issue: #16851

**Comment by ashiagr** [comment]
Closing in favor of https://github.com/wordpress-mobile/WordPress-Android/pull/16851.

**Pull request closed without merging**